### ISSUE TITLE
Use of uninitialized value $t_g2[1] in hash element at C2_filter_and_select_best_EX_matches_by_targetgene.pl line 70, <INONE> line 2.

### ISSUE THREAD

**Issue by Yuzhang0102** [biocorecrg/ExOrthist#31]
The log file said "Missing output file(s) `PART_*-cluster_input.tab` expected by process `format_EX_clusters_input`".
But in the filter and select best EX matches step, the C2 script report errors nearly in each line...And the filtered_best_scored_EX_matches_by_targetgene.tab is empty. The error is :
```
Use of uninitialized value $t_g2[1] in hash element at ~/bioapps/ExOrthist-master/bin/C2_filter_and_select_best_EX_matches_by_targetgene.pl line 70, <INONE> line 2.
Use of uninitialized value $t_g2[1] in hash element at ~/bioapps/ExOrthist-master/bin/C2_filter_and_select_best_EX_matches_by_targetgene.pl line 71, <INONE> line 2.
Use of uninitialized value $t_g2[1] in hash element at ~/bioapps/ExOrthist-master/bin/C2_filter_and_select_best_EX_matches_by_targetgene.pl line 72, <INONE> line 2.
``` 
Look forward to your reply

**Comment by alessiovignoli** [comment]
Hello, 

do not now if it is still relevant, but I encountered a similar error message:
`Use of uninitialized value $l[2] in hash element at ExOrthist/bin/B5_format_aln_info_by_best_isoform_match.pl line 29, <IN> line 2.`

It was caused by the fact that my filenames were the following, example:
`--genomes Homo-sapiens.fasta  --annotations Homo-sapiens.gtf`
the --cluster:
`GF0001 Homo-sapiens ENSG00001`
and --evodists:
`Homo-sapiens Mus-musculus short`

Basically it seems that the identifier 'Homo-sapiens' (in this case) should be only alphanumerical digits, stuff like '_', '-', ecc. could create troubles and errors. Only tried _ and -     :)

hope it can help.
